pkgs/development/compilers/haxe/default.nix

```diff
@@ -1,21 +1,29 @@
-{ stdenv, fetchgit, ocaml, zlib, pcre, neko, camlp4 }:
-
-stdenv.mkDerivation {
-  name = "haxe-3.4.2";
+{ stdenv, fetchgit, fetchpatch, ocaml, zlib, pcre, neko, camlp4 }:
+
+let
+  # Unlike PERL5LIB and CLASSPATH, HAXELIB_PATH allows to specify only one location, making setupHook disfunctional
+  # An attempt to fix it in upstream has been failed: https://github.com/HaxeFoundation/haxelib/pull/206
+  multiple-repositories-on-HAXELIB_PATH = fetchpatch {
+    url    = https://github.com/volth/haxelib/compare/dd0ad7cf3a8e3fa8e3765928364a55839cabe06f.patch;
+    sha256 = "0g1vrhfa901im20zjd45vbbm2j72rw0vjy9apn7vacdak68h4apy";
+  };
```

Comment:
@Mic92, I called that "fail" because the only reviewer appeared when that PR was 8 month old. Now there is more activity, so we could wait until the patch is accepted by upstream and pick the patch from the upstream repo instead of mine.